nsq/deflate_socket.py

```diff
@@ -38,8 +38,14 @@ def _recv(self, size, method):
         return uncompressed
 
     def send(self, data):
-        if isinstance(data, memoryview):
```

Comment:
I think we want to keep `memoryview` handling added in #191

Reply:
(or this logic can be moved into AsyncConn?)

Reply:
@jehiah Well, I guess it is not necessary anymore.
Since we compress the data in the layer above (before sending the data to the IOStream.write) it's no longer "our" responsibility, because compression will return just another bytestring which will be sent to `stream.write` and through tornado's functions calls untill the `socket.send` will eventually send it. 

https://github.com/tornadoweb/tornado/blob/master/tornado/iostream.py#L556 
Tells us that data might be both memoryview and bytes, so not a big deal if we send bytes.

The `send` method receives the result of `_command` call from https://github.com/nsqio/pynsq/blob/master/nsq/protocol.py#L82 
therefore it always will be bytes.

Please, tell me if I'm missing some cases which might bring the memoryview type into the compression level.

Reply:
I think that the problem fixed in #191 is that the compressor did not handle a `memoryview`, the compressor required bytes, and depending on where the data came from (maybe another tornado read) it might be a memoryview. So we may need to convert memoryview to bytes before passing it to the compressor, either in `AsyncConn`:

```python
    def send(self, data):
        self.stream.write(self.encoder.encode(data))
```

or maybe just in the compressing encoders e.g. `SnappyEncoder`:

```python
    def encode(data):
        return snappy.StreamCompressor().compress(data)
```

Reply:
... I'll admit I'm not sure how `AsyncConn.send()` got a `memoryview` before if it always gets the result of `_command()`, but #191 says it did ...

Reply:
... oh I see, the previous abstraction was a Socket we give to a tornado IOStream, but now the compression is outside of that whole thing ... ok I see how this is all fine